src/components/data/QueryAuth0.js

```diff
@@ -38,100 +40,117 @@ export default class QueryAuth0 extends React.PureComponent {
     userProfileReceived: PropTypes.func.isRequired,
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
   async componentDidMount() {
-    const { authenticationFailed } = this.props;
+    this.checkForAuthentication();
+
+    window.addEventListener('focus', this.checkForAuthentication);
+  }
+
+  componentDidUpdate(prevProps) {
+    const { accessToken } = this.props;
+
+    if (accessToken && accessToken !== prevProps.accessToken) {
+      // Check the access token now
+      this.validateAccessToken();
+
+      // Set up periodic checks of the access token
+      this.setState(() => ({
+        valideAccessTokenInterval: window.setInterval(
+          this.validateAccessToken,
+          CHECK_AUTH_EXPIRY_INTERVAL_MS,
+        ),
+      }));
+    } else if (!accessToken) {
+      // There is no access token so stop periodic checks
+      this.clearValidateAccessTokenInterval();
+    }
+  }
+
+  componentWillUnmount() {
+    this.clearValidateAccessTokenInterval();
+  }
+
+  async checkForAuthentication() {
+    const { accessToken, authenticationFailed } = this.props;
     let authResult;
+    let expiresAt;
 
+    // Check whether we are currently in an auth flow
     try {
       authResult = await parseHash();
     } catch (err) {
       authenticationFailed(err);
     }
 
-    // By default this is true if the window.location.hash parsing stuff worked.
-    // We might change our minds about this later.
-    let startAccessTokenRefreshLoop = !!authResult;
-
+    // If not check if the user has a locally stored authResult
     if (!authResult) {
-      // It if was null from finishAuthenticationFlow() it means it didn't pick anything
-      // up from the window.location.hash so we can potentially use what was stored in
-      // localStorage.
-      authResult = JSON.parse(localStorage.getItem('authResult'));
-      if (authResult) {
-        // If you have previously logged in we'll want to start the accessToken refresh
-        // loop. Even if it has expired. Because, if it indeed as expired, the
-        // accessToken refresh loop can silently authenticate you again.
-        startAccessTokenRefreshLoop = true;
-      }
-      // But this is only good enough if it hasn't expired.
-      const expiresAt = JSON.parse(localStorage.getItem('expiresAt'));
-      if (expiresAt && expiresAt - new Date().getTime() < 0) {
-        // Oh no! It has expired.
-        authResult = null;
+      expiresAt = JSON.parse(localStorage.getItem('expiresAt'));
```

Comment:
I know this is the prevailing style, but shouldn't we be prefixing the local storage key names with something like `auth`? It seems weird that we have a key just named `expiresAt` in a site-global store.